sunpy/cm/__init__.py

```diff
@@ -1,7 +1,28 @@
-"""
-The following colormaps are provided by this module:
-"""
-from sunpy.cm.cm import *  # noqa
+import sys as _sys
 
-for cmname in cmlist.keys():  # noqa
-    __doc__ += f"\n* '`~sunpy.cm.cm.{cmname}`'\n"
+# Deprecate the module without polluting the namespace
+import warnings as _w
+from sunpy.util.exceptions import SunpyDeprecationWarning as _SDW
+
+_w.warn("The functionality of the sunpy.cm module is now in sunpy.visualization.colormaps as of SunPy 1.1. "
+        "The ability to import sunpy.cm as a module may be removed in a future version of SunPy.",
+        category=_SDW)
+
+# Import docstring from the new module
+import sunpy.visualization.colormaps
+
+__doc__ = sunpy.visualization.colormaps.__doc__
+
+# Trick Python into being able to do from sunpy.cm.cm import sdoaia171
+# This does mean these two are imported by default where only cm was before.
+from sunpy.visualization.colormaps import color_tables
+from sunpy.visualization.colormaps import cm
+
+_sys.modules['sunpy.cm.cm'] = sunpy.visualization.colormaps.cm
+_sys.modules['sunpy.cm.color_tables'] = sunpy.visualization.colormaps.color_tables
+
+# Import the two original functions in this namespace and use __all__ to keep
+# import * behaviour the same.
+from sunpy.visualization.colormaps.cm import *
+
+__all__ = ['show_colormaps', 'cmlist']
```

Comment:
Technically, you're changing the behavior of `from sunpy.cm import *`, but that's because it didn't previously have an `__all__`.  (For example, `cmname` would get imported.)  However, the previous behavior is essentially a bug, and no one would have relied on it, so I support this change.

Reply:
I think it is only cname.

This `__all__` is copied from `sunpy.cm.cm.__all__` which we were importing [with import *](https://github.com/sunpy/sunpy/blob/master/sunpy/cm/__init__.py#L4)